tests/test_transforms/test_timestamp/test_holiday_transform.py

```diff
@@ -224,3 +320,52 @@ def test_save_load(example_tsds):
 def test_params_to_tune():
     transform = HolidayTransform()
     assert len(transform.params_to_tune()) == 0
+
+
+@pytest.mark.parametrize(
+    "freq, expected_result",
+    (
+        ("Q-JAN", [pd.Timestamp("1999-11-01 00:00:00"), pd.Timestamp("2000-01-31")]),
+        ("M", [pd.Timestamp("2000-01-01 00:00:00"), pd.Timestamp("2000-01-31 00:00:00")]),
+        ("W-MON", [pd.Timestamp("2000-01-31 00:00:00"), pd.Timestamp("2000-02-06 00:00:00")]),
+    ),
+)
+def test_define_period_check_end(freq, expected_result):
```

Comment:
Let's make one large test with all our test cases. There should be frequencies: "W", "W-TUE", "M", "MS", "Q", "Q-NOV", "QS", "QS-NOV", "Y", "YS", "A-OCT", "AS-OCT".